### DISCUSSION TITLE
HM-PB-2-WM will not update state in HA

### DISCUSSION THREAD

**Discussion by privat99** [SukramJ/aiohomematic#1820] [General]
### I agree to the following

- [X] I have read the [documentation](https://github.com/danielperna84/custom_homematic#custom_homematic) right now
- [X] I have read the [FAQs](https://github.com/danielperna84/custom_homematic#frequently-asked-questions) right now
- [X] I am aware of the latest [release notes](https://github.com/danielperna84/custom_homematic/releases)
- [X] I am aware that an integration that was installed via HACS also needs to be updated via HACS
- [X] The Backend (CCU/Homegear/...) is working as expected, devices are taught in and are controllable by its own UI.
- [X] I am running the latest version of the custom_component (and Home Assistant)

### The problem

I want to use HM-PB-2-WM in HomeAssistant.
The state will not been updated.
![grafik](https://github.com/user-attachments/assets/84266fef-f1d0-4c53-b145-2e2b46d71bb6)
Some seconds before, I pressed both button long an short.
This bill not be shown in HA.
The CCU3 the key presses are shown:
![grafik](https://github.com/user-attachments/assets/1de7ec2f-1426-4e8f-b9d1-302afa9e75a6)

{
  "home_assistant": {
    "installation_type": "Home Assistant OS",
    "version": "2024.10.4",
    "dev": false,
    "hassio": true,
    "virtualenv": false,
    "python_version": "3.12.4",
    "docker": true,
    "arch": "aarch64",
    "timezone": "Europe/Berlin",
    "os_name": "Linux",
    "os_version": "6.6.31-haos-raspi",
    "supervisor": "2024.10.3",
    "host_os": "Home Assistant OS 13.2",
    "docker_version": "27.2.0",
    "chassis": "embedded",
    "run_as_root": true
  },
  "custom_components": {
    "homematicip_local": {
      "documentation": "https://github.com/danielperna84/custom_homematic",
      "version": "1.68.1",
      "requirements": [
        "hahomematic==2024.10.14"
      ]
    },

Please can you help me?

Thanks

### What version of HomematicIP (local) has the issue?

2024.10.14

### What type of installation are you running?

Home Assistant OS

### What type of installation are you running for your homematic backend?

CCU3

### Which version of your homematic backend are you running?

3.79.6

### What hardware are you running for your system?

PI4

### Which config details do you use

- [ ] TLS
- [ ] callback data ([see](https://github.com/danielperna84/custom_homematic#callback_host-and-callback_port))

### Which interfaces do you use?

- [X] Homematic IP
- [X] Homematic (Bidcos-RF)
- [ ] Groups (Heating-Group)
- [ ] BidCos-Wired (HM-Wired)

**Comment by SukramJ** [comment]
Current version is 1.69.0. Please update.
Did you read the docs? https://github.com/danielperna84/custom_homematic#events-for-homematicip-devices

**Comment by privat99** [comment]
Thanks for your quick response.
I have made the Update to 1.70.0
The CCU program is installed
![image](https://github.com/user-attachments/assets/cd00b6ad-4233-48c4-82fe-66b75bdfcc62)
After setup I have started it.
Then I pressed the buttons
![image](https://github.com/user-attachments/assets/d1f8d841-ab2d-440b-9400-956bd28a09fc)
In HA nothing happend
![image](https://github.com/user-attachments/assets/bca73442-d14e-4666-96fa-95e672ab6559)


Where is my fault?

**Reply by SukramJ** [reply]
Das CCU-Programm ist aktiv und wird beim Drücken einer Taste ausgelöst?
Die Konfiguration wurde auf das Gerät übertragen? -> keine offenen Servicemeldungen
Richtiges Gerät in HA ausgewählt?
Gibt es Meldungen im HA Log(full log)?

**Comment by SukramJ** [comment]
1.70.0 is beta, so it is your own risk!

**Comment by privat99** [comment]
Hallo,

ich stehe auf dem Schlauch.
Ich habe das Programm angelegt. Alle Schalterkombinationen in kurz und lang angelegt.
Danach kein weiteren Schritte im Programm konfiguriert.
![image](https://github.com/user-attachments/assets/4af2c7ab-9f6c-4e8b-b7a9-fc4df361e424)

Danach hatte ich die Meldung, dass für den Taster eine Konformationsänderung ansteht. Diese bestätigt.

Was mich wundert ist, dass das Programm bei einem Tastendruck nicht ausgeführt wird.

Hast Du für mich einen Tipp was ich falsch mache?
Danke

**Reply by SukramJ** [reply]
Sieht eigentlich gut aus.
Warum hast du nicht den Service aufgerufen, anstatt ein Programm anzulegen?

**Comment by privat99** [comment]
Was meinst Du mit Service aufrufen?
Ich stelle fest, dass ich ein DAU bin. Sorry

**Reply by SukramJ** [reply]
> Current version is 1.69.0. Please update.
Did you read the docs? https://github.com/danielperna84/custom_homematic#events-for-homematicip-devices

Steht im Link! Option A

**Comment by privat99** [comment]
Danke für Deine Hilfe. Ich komme nicht weiter.
Ich habe versucht das mit dem Programm durchzuführen, da ich den Central_Link nicht verstanden habe, wie und was ich machen muss.
Scheinbar ist dort mein Fehler

**Reply by SukramJ** [reply]
Ob Serviceaufruf oder Programm ist im Ergebnis das Gleiche. Der Serviceaufruf ist halt einfacher.

**Comment by privat99** [comment]
Hallo,

wenn ich eine leere Batterie in den Schalter hineinlege, dann wird der niedrige Batteriestand übermittelt.
Die Tastenbefehle nicht.

Ich mache etwas falsch. Was verstehe ich nicht.
Das mit dem Serviceaufruf habe ich nicht verstanden. Was muss ich dafür machen und was sind die Schritte.

**Reply by SukramJ** [reply]
> Ich mache etwas falsch. Was verstehe ich nicht.
Das mit dem Serviceaufruf habe ich nicht verstanden. Was muss ich dafür machen und was sind die Schritte.

Egal, das Programm ist ausreichend.
Warum, das bei dir nicht funktioniert kann ich dir leider nicht beantworten.
@Baxxy13 hast Du eine Idee?

**Reply by Baxxy13** [reply]
Hmm, nicht so wirklich.
Ich kann das auch nicht reproduzieren.

Habe einen HM-PB-2-WM55 (FW: 1.4) aus der Testgerätekiste gefischt und angelernt.
Ansonsten nix gemacht, also weder Programm noch Service ausgeführt.
Nach Neustart von HA wurde das Teil gefunden und ich sehe alle Ereignisse wenn ich auf den Tasten des HM-PB-2-WM55 rumdrücke.

**Reply by SukramJ** [reply]
@Baxxy13 Danke.

@privat99 Vielleicht kannst du das mal versuchen: Gerät von der CCU ablernen und wieder anlernen.

**Comment by privat99** [comment]
Hallo,

habe den Fehler gefunden. Die Übertragung stand auf gesichert.
Sorry, dass ich so genervt habe. 
Danke für Eure Hilfe

**Reply by Baxxy13** [reply]
Gesicherte Übertragung war bei mir auch aktiv, trotzdem ging das. 
Ich habe aber schon mal irgendwo gelesen das es damit manchmal Probleme gibt. 
Aber gut das es nun funktioniert.
